python/plugins/processing/core/GeoAlgorithm.py

```diff
@@ -215,7 +215,8 @@ def execute(self, progress=SilentProgress(), model=None):
             lines = [self.tr('Uncaught error while executing algorithm')]
             lines.append(traceback.format_exc())
             ProcessingLog.addToLog(ProcessingLog.LOG_ERROR, lines)
-            raise GeoAlgorithmExecutionException(unicode(e) + self.tr('\nSee log for more details'), lines, e)
+            raise GeoAlgorithmExecutionException(
+                unicode(e.message, errors='replace') + self.tr('\nSee log for more details'), lines, e)
```

Comment:
Ok, let's merge that for 2.18 and see that we get proper unicode support everywhere in QGIS 3.0